orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/Front50Service.groovy

```diff
@@ -54,8 +54,8 @@ interface Front50Service {
   @GET("/pipelines/{applicationName}")
   List<Map<String, Object>> getPipelines(@Path("applicationName") String applicationName, @Query("refresh") boolean refresh)
 
-  @GET("/pipelines/{pipelineId}/history")
-  List<Map<String, Object>> getPipelineHistory(@Path("pipelineId") String pipelineId, @Query("limit") int limit)
+  @GET("/pipelines/{pipelineId}/get")
```

Comment:
Agreed on the endpoint endpoint, but was forced to do that by the fact that we already defined the list operation as `/pipelines/{applicationName}` and the history operation as `/pipelines/{pipelineId}/history`...so there weren't really any other unambiguous endpoints available.

It's true that this could have short-circuited the `history` endpoint, but this seems like a fundamentally different type of request...it seems strange to me from an API design standpoint that the much more common operation of "get the current pipeline" would need to go to a history endpoint and pass `limit=1`.  (Particularly given that there are pretty different performance requirements between getting the current pipeline and getting the history.)

Reply:
I think the API design was made convoluted by some historical choice of `/pipelines/{applicationName}` -- the pattern was consistent enough across all resources that `/history?limit=1` became a thing.

For some controllers ... we've started new v2 controllers (`/v2/{resource}`) that behave more along the lines of what we would expect. 

Adding a `/get` is almost certainly going to be short-lived in my opinion. I'd rather just optimize the `?limit=1` and plan for a better api across all resources in `front50`.

Thoughts from @robzienert?

Reply:
:man_shrugging: I'm not sure I see a strong reason to hold off on adding this endpoint (or rather a reason to remove it given that the front50 change was merged a couple of weeks ago).

I definitely think that adding `/v2/{}` endpoints that have a more standard structure is a good idea, but I don't think that the plan to eventually do this needs to preclude adding this useful endpoint to the existing API.  Even if we do write a `/v2/` API with a reasonable GET operation (that presumably returns just a single item or 404s) we'll need to update callers to use that endpoint, changing them from expecting a List to expecting a single item, which is the majority of what this PR does.  Moving to the v2 endpoint would just be a change to the `Front50Service` at that point, so it's not like this is wasted work or churn---it's still moving us towards where we want to be.

This endpoint will also avoid having new code written against the history endpoint...the [performance issue](https://github.com/spinnaker/spinnaker/issues/5085#issuecomment-562125988) I'm fixing was fairly recently introduced (spinnaker/echo#518) and I'd like to avoid more issues like that.

We could optimize the history endpoint...it seems like it has caused issues in the past that were unfortunately only fixed for one storage provider rather than more generally (spinnaker/front50#316).  But it really still seems strange to me that we don't have a normal get operation on this API and it's not clear to me why it's controversial to add one.

@plumpy reviewed the [PR adding this endpoint](https://github.com/spinnaker/front50/pull/644), would value his feedback on this as well.

Reply:
Yeah, it's pretty normal to have a `get` API... the fact that we don't have a great endpoint available to host that API is definitely a failing of the so-called `v1` APIs, but I can't see why that should preclude us from adding one at an admittedly lame URL. If the rest of the API makes sense and the name is the only problem, but we literally don't have a way to use a better name, then :man_shrugging:

Reply:
Full circle -- not disputing the need for `GET`, that's obviously standard and should have been there all along.

Just calling out distaste for the slightly lame URL. 

Merge away.

Reply:
Sounds good, thanks!